metric_learn/rca.py

```diff
@@ -26,7 +26,11 @@
 def _chunk_mean_centering(data, chunks):
   num_chunks = chunks.max() + 1
   chunk_mask = chunks != -1
-  chunk_data = data[chunk_mask]
+  # Warning: we need to ensure we don't overwrite the data
+  # through slices hence we do a copy. We will also need to
+  # ensure the data is float so that we can substract the
+  # mean on it
+  chunk_data = data[chunk_mask].astype(float, copy=True)
```

Comment:
Logical indexing always creates a copy, so the `copy=True` isn't necessary here. In fact, `astype()` defaults to returning a new copy each time, so what we really want here is `copy=False` to avoid unnecessary copies.

Reply:
That's right, thanks